webapp/app/appRouterSwitch.js

```diff
@@ -1,55 +1,44 @@
 import React, { useEffect } from 'react'
-import { connect } from 'react-redux'
+import { useDispatch, useSelector } from 'react-redux'
 import { Route, Switch } from 'react-router-dom'
-import { initApp } from './actions'
 
-import DynamicImport from '../components/dynamicImport'
-import LoginView from '../login/loginView'
+import DynamicImport from '@webapp/components/dynamicImport'
+import LoginView from '@webapp/login/loginView'
 import Loading from '@webapp/components/loading'
 
 import * as AppState from '@webapp/app/appState'
-import * as UserState from '@webapp/user/userState'
 
-const AppRouterSwitch = props => {
-  const { loggedIn, initApp, applicationStatus } = props
+import { initApp } from './actions'
+
+const AppRouterSwitch = () => {
+  const dispatch = useDispatch()
+  const appStatus = useSelector(AppState.getApplicationStatus)
 
   useEffect(() => {
-    initApp()
+    dispatch(initApp())
```

Comment:
Can we awoid using dispatch directly inside the _view_?

Reply:
here I use the ```useDispatch()``` hook to avoid the connect...
it's the same